src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForGet(index);
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current != null ? current.value : null;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexForGet(index);
+        Node<T> current = getNode(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexForGet(index);
+        Node<T> current = getNode(index);
+        T removeValue = current.value;
+        if (current == first && size > 1) {
+            current.next.prev = null;
+            first = current.next;
+        } else if (current == last && size > 1) {
+            current.prev.next = null;
+            last = current.prev;
+        } else if (size > 1) {
+            current.next.prev = current.prev;
+            current.prev.next = current.next;
+        }
+        size--;
+        return removeValue;
     }
 
     @Override
     public boolean remove(T object) {
+        int indexObject = searchIndex(object);
+        if (indexObject > -1) {
+            remove(indexObject);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void linkLast(T value) {
+        Node<T> newNode = new Node<T>(last, value, null);
+        last.next = newNode;
+        last = newNode;
+    }
+
+    private void linkFirst(T value) {
+        Node<T> newNode = new Node<T>(null, value, first);
+        first = newNode;
+        last = newNode;
+    }
+
+    private boolean checkIndexForAdd(int index) {
+        if (index > size() || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " is bad for size " + size());
+        }
+        return true;
+    }
+
+    private boolean checkIndexForGet(int index) {
+        if (index >= size() || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " is bad for size " + size());
+        }
+        return true;
+    }
+
+    private int searchIndex(T object) {
+        Node<T> current = first;
+        for (int i = 0; i < size; i++) {
+            if (current.value == object || current.value != null && current.value.equals(object)) {
+                return i;
+            }
+            current = current.next;
+        }
+        return -1;
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> current = first;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
```

Comment:
Please recheck checklist
https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#dont-iterate-your-nodes-from-the-head-to-the-tail-or-vice-versa